src/containers/Assign/useGetAssignmentOptions.ts

```diff
@@ -0,0 +1,108 @@
+import { AssignmentDetails, PageElement, User } from '../../utils/types'
+import {
+  ReviewAssignmentStatus,
+  ReviewStatus,
+  TemplateElementCategory,
+} from '../../utils/generated/graphql'
+import { AssignmentOption } from '../../utils/data/assignmentOptions'
+import { useLanguageProvider } from '../../contexts/Localisation'
+
+const useGetAssignmentOptions = () => {
+  const { strings } = useLanguageProvider()
+  const getOptionFromAssignment = ({
+    review,
+    reviewer,
+    isCurrentUserReviewer,
+  }: AssignmentDetails) => ({
+    key: reviewer.id,
+    value: reviewer.id,
+    text: isCurrentUserReviewer
+      ? strings.ASSIGNMENT_YOURSELF
+      : `${reviewer.firstName || ''} ${reviewer.lastName || ''}`,
+    disabled: review?.current.reviewStatus === ReviewStatus.Submitted,
+  })
+
+  interface GetAssignmentOptionsProps {
+    assignments: AssignmentDetails[]
+    sectionCode: string
+    elements: PageElement[]
+    previousAssignee?: number
+  }
+
+  const getAssignmentOptions = (
+    { assignments, sectionCode, elements, previousAssignee }: GetAssignmentOptionsProps,
+    currentUser: User | null
+  ) => {
+    const currentSectionAssignable = assignments.filter(
+      ({ assignableSectionRestrictions }) =>
+        assignableSectionRestrictions.length === 0 ||
+        assignableSectionRestrictions.includes(sectionCode)
+    )
+
+    const currentUserAssignable = currentSectionAssignable.filter(
+      (assignment) => assignment.isCurrentUserAssigner
+    )
+
+    // Dont' want to render assignment section row if they have no actions
+    if (currentUserAssignable.length === 0) return null
+    const numberOfAssignableElements = elements.filter(
+      ({ element }) =>
+        (!sectionCode || element.sectionCode === sectionCode) &&
+        element.category === TemplateElementCategory.Question
+    ).length
+
+    if (numberOfAssignableElements === 0) return null
```

Comment:
So is this what's causing https://github.com/openmsupply/application-manager-web-app/issues/1020 ?

I thought elements would still be assignable, even for questions that don't have a response? Like, when generating review-question-assignments, we're not considering applicant responses, are we?

Reply:
I think no, we only create question assignments when there is a response given by the applicant. But would need to check. Also, don't think that's the cause of that bug. Actually, the problem is allowing to select a reviewer for a section if a non-element is reviewable. And I believe this will be changed once we define how to implement the solution required by Jess to prevent only selecting the decision based on review per element or allow review per sections.